src/frameworks/CoreServices/include/CarbonCore/Multiprocessing.h

```diff
@@ -14,6 +14,8 @@ enum {
 };
 
 typedef void* MPCriticalRegionID;
+typedef void* MPSemaphoreID;
```

Comment:
While you could have it just be a `void*` type, I think it would be better if we follow how Apple does it and make it an [opaque pointer](https://interrupt.memfault.com/blog/opaque-pointers).

https://developer.apple.com/documentation/coreservices/mpsemaphoreid
```suggestion
typedef struct OpaqueMPSemaphoreID *MPSemaphoreID;
```

Reply:
Shouldn't MPCriticalRegionID be `typedef struct OpaqueMPCriticalRegionID *MPCriticalRegionID;typedef struct OpaqueMPCriticalRegionID *MPCriticalRegionID;  then? it's the only reason I did it this way

Reply:
I didn't see this comment. Yeah, it should probably be changed as well.